### PR TITLE
Add format option for fields.time #686

### LINKED ISSUE #686: Add format option for fields.time
ISO 8601 supports following formats:
> hh:mm:ss.sss or hhmmss.sss
> hh:mm:ss or hhmmss
> hh:mm or hhmm
> hh

It would be nice to have a format option for fields.time, like 
`begin = fields.Time(format='hh:mm')`

### PR DESCRIPTION
**Pull request by infinityxxx**
Closes #686

### CODE REVIEW

File: src/marshmallow/fields.py [hunk 1]
```diff
@@ -1314,34 +1314,29 @@ def _deserialize(self, value, attr, data, **kwargs):
         return ret
 
 
-class Time(Field):
-    """ISO8601-formatted time string.
+class Time(DateTime):
```
Comment (sloria) [inline src/marshmallow/fields.py:1317]: thought (non-blocking): It feels a bit odd to make `Time` a subclass of `DateTime`.

But I suppose this is consistent with `Date` being a subclass of `DateTime`.

**Review by sloria — COMMENTED** [review]
Thanks for the PR!

I only did a cursory glance for now. I'll try to review this more thoroughly within the coming weeks.

**Comment by infinityxxx** [comment]
@sloria I'm participating in hacktoberfest and would really appreciate if this PR could qualify for that (approved before October ends / labeled as "hacktoberfest-accepted"), I'll do my best to make the required changes if needed! I can also take on more issues :D

**Review by sloria — APPROVED** [review]
Looks good! I'll merge and release this over the weekend.

**Pull request merged**